### ISSUE TITLE
Does not handle Unix convention of concatenated options

### ISSUE THREAD

**Issue by cxj** [auraphp/Aura.Cli#58]
`Context::getopt()` (probably) doesn't allow for the Unix convention of shortening options by concatenating them.  Example:
```sh
896 pds/skeleton$ tar -c -v -f foo.tar src/
a src
a src/ComplianceValidator.php
a src/Console.php
a src/PackageGenerator.php
a src/TestRunner.php


897 pds/skeleton$ tar -cvf foo.tar
src/
src/ComplianceValidator.php
src/Console.php
src/PackageGenerator.php
src/TestRunner.php
```
Note how first call to `tar` uses separated arguments `-c` `-v` and `-f foo.tar`, compared to second call where `-cvf` are concatenated.  Most *nix commands follow this convention.

However when using Aura.Cli with the latter (concatenated) case, `$getopt->get('-f')` returns `1` instead of `foo.tar`.

**Comment by pmjones** [comment]
Huh, I thought it *did* allow for that. I'll check to make sure.

**Comment by cxj** [comment]
I was surprised myself when it did not.  Do you need an example?

**Comment by pmjones** [comment]
OK, it does allow for a "cluster" of options, cf. <https://github.com/auraphp/Aura.Cli/blob/2.x/tests/Context/GetoptParserTest.php#L281-L312>

What I have *not* tested is that if the final option in a concatenation, if it takes an argument, it should presume that the next argument belongs to it.

Would you be OK with writing a test for that, and maybe sending along a related fix, as a PR?

**Comment by cxj** [comment]
Here is the option input array I'm using:
```php
$this->getopt = $context->getopt([
    0 => 'v,verbose',
    1 => 'h,help',
    2 => 'c::',
    3 => '#action?',
]);
```
I call my CLI program with `generate -vc some/dir/path`.

**Comment by cxj** [comment]
I'll take a look at writing a test and PR.

**Comment by cxj** [comment]
I've been working on finding a fix using the above #60 test for hours now, without success.  Either I'm really not understanding what's going on, or the code is seriously wrong and just accidentally works in most cases.  For example, this bit of code in `setShortFlagValues()` seems completely opposite of what's needed to pick up value arguments required or optional:
```php
          if (! $this->shortOptionRequiresValue($option, $name)) {
                $this->setValue($option, true);
          }
```
Yet if I reverse the sense, the option _flags_ which expect to be `true`if present, `false` if not, do not get set to `1` which is then regarded as `true` later in the code.  At the same time, the option does not get the value, but instead gets `true` and a generated index (e.g. `0` below) somehow grabs the option value:
```
Failed asserting that Array &0 (
    '-b' => true
    0 => 'foo_arg'
) is identical to Array &0 (
    '-f' => true
    '-z' => true
    '-b' => 'foo_arg'
)
```

If instead, I change the code by adding a recursive call to `setShortFlagValue()`, as shown below, I get the right data out, but `::parseInput()` call returns `false`, i.e. an error.
```php
            if (! $this->shortOptionRequiresValue($option, $name)) {
                $this->setValue($option, true);
            }
            $this->setShortFlagValue($name);
```

The `false` value is because this function seems to _always set an error_ whenever a short option requires a value.  How does that make sense?
```php
    protected function shortOptionRequiresValue($option, $name)
    {
        if ($option->param == 'required') {
            $this->errors[] = new Exception\OptionParamRequired(
                "The option '$name' requires a parameter."
            );
            return true;
        }
        return false;
    }
```

**Comment by kenjis** [comment]
I believe #60 and #67 fixed this issue.
